File: 560-subarray-sum-equals-k/subarray-sum-equals-k.go
func subarraySum(nums []int, k int) int {
    
    count :=0
    prefixsum :=0
    prefixcount := make(map[int]int)
    prefixcount[0] = 1

    for _, num := range nums {

        prefixsum += num

        if freq, exists := prefixcount[prefixsum - k]; exists{
            count += freq
        }

        prefixcount[prefixsum] +=1
    }    

    return count
}

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution:

    def subarraySum(self, nums: List[int], k: int) -> int:
        count = 0
        prefixsum = 0
        prefixcount = {0: 1}

        for num in nums:
            prefixsum += num

            # Checks if (prefixsum - k) exists in the dictionary and adds its frequency
            if (prefixsum - k) in prefixcount:
                count += prefixcount[prefixsum - k]

            # Python's dict.get() handles missing keys safely (defaulting to 0)
            prefixcount[prefixsum] = prefixcount.get(prefixsum, 0) + 1

        return count